### PR TITLE
[FEATURE] Add support of user

### PR DESCRIPTION
**Pull request by Nexucis**
This PR is adding the support of a new resource "User".
This should help for the authentication.

### CODE REVIEW

File: internal/api/impl/v1/user/service.go [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+		return nil, updateErr
+	}
+	return v1.NewPublicUser(entity), nil
+}
+
+func (s *service) Delete(parameters shared.Parameters) error {
+	return s.dao.Delete(parameters.Name)
+}
+
+func (s *service) Get(parameters shared.Parameters) (interface{}, error) {
```
Comment (celian-garcia) [inline internal/api/impl/v1/user/service.go:117]: The dao return a User. Does it mean that the get will return the password ?

File: internal/api/impl/v1/user/service.go [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+	return s.dao.Get(parameters.Name)
+}
+
+func (s *service) List(q databaseModel.Query, _ shared.Parameters) (interface{}, error) {
```
Comment (celian-garcia) [inline internal/api/impl/v1/user/service.go:121]: Same for List will it return the password ?

File: pkg/client/api/v1/user.go [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+	client *perseshttp.RESTClient
+}
+
+func newUser(client *perseshttp.RESTClient) UserInterface {
+	return &user{
+		client: client,
+	}
+}
+
+func (c *user) Create(entity *v1.User) (*v1.User, error) {
```
Comment (celian-garcia) [inline pkg/client/api/v1/user.go:49]: AFAIU the returns should all be PublicUser no ?

**Review by celian-garcia — CHANGES_REQUESTED** [review]

File: pkg/model/api/v1/public_user.go [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+package v1
+
+import (
+	modelAPI "github.com/perses/perses/pkg/model/api"
+	"github.com/perses/perses/pkg/model/api/v1/secret"
+)
+
+type PublicUserSpec struct {
+	FirstName string        `json:"firstName,omitempty" yaml:"firstName"`
```
Comment (Gladorme) [inline pkg/model/api/v1/public_user.go:22]: `omitempty` missing for yaml, is it normal or is it a trick for something? 👀 
If it's intended maybe explain a bit in a comment 😄

File: pkg/model/api/v1/public_user.go [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+func NewPublicUserSpec(u UserSpec) PublicUserSpec {
+	return PublicUserSpec{
+		FirstName: u.FirstName,
+		LastName:  u.LastName,
+		Password:  secret.Hidden(u.Password),
+	}
+}
+
+type PublicUser struct {
```
Comment (Gladorme) [inline pkg/model/api/v1/public_user.go:35]: No yaml?

File: pkg/model/api/v1/user.go [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+
+import (
+	"encoding/json"
+	"fmt"
+
+	modelAPI "github.com/perses/perses/pkg/model/api"
+)
+
+type UserSpec struct {
+	FirstName string `json:"firstName,omitempty" yaml:"firstName"`
```
Comment (Gladorme) [inline pkg/model/api/v1/user.go:24]: `omitempty` missing for yaml, is it normal or is it a trick for something? 👀 
If it's intended maybe explain a bit in a comment 😄

File: pkg/model/api/v1/user.go [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+	LastName  string `json:"lastName,omitempty" yaml:"lastName"`
+	Password  []byte `json:"password,omitempty" yaml:"password"`
+}
+
+func (p *UserSpec) UnmarshalJSON(data []byte) error {
+	type plain = struct {
+		FirstName string `json:"firstName,omitempty"`
```
Comment (Gladorme) [inline pkg/model/api/v1/user.go:31]: No yaml?

File: internal/api/impl/v1/user/service.go (hunk 1 above)
Comment (Nexucis) [reply internal/api/impl/v1/user/service.go:117]: ah yeah I miss this part. That wasn't intended. I will fix that

File: pkg/model/api/v1/public_user.go (hunk 4 above)
Comment (Nexucis) [reply pkg/model/api/v1/public_user.go:22]: oups miss that too, will fix it

File: pkg/client/api/v1/user.go (hunk 3 above)
Comment (Nexucis) [reply pkg/client/api/v1/user.go:49]: it does't really matter for the client. `PublicUser` on the API side is just here to change the behaviour of the `Marshalling` when the document is going outside of Perses (database included)

**Review by celian-garcia — APPROVED** [review]

**Pull request merged**